src/cascadia/TerminalSettingsModel/ActionMap.cpp

```diff
@@ -454,156 +475,73 @@ namespace winrt::Microsoft::Terminal::Settings::Model::implementation
-                    maskingCmd = *inheritedCmdImpl->Copy();
-                    _MaskingActions.emplace(actionID, maskingCmd);
-                }
-            }
-        }
-        else
-        {
-            // This is an action that we already have a mutable "masking" record for.
-            // Record it so that we update it with any new information.
-            maskingCmd = maskingActionPair->second;
-        }
-    }
-
-    // Method Description:
-    // - Update our internal state with the name of the newly registered action
-    // Arguments:
-    // - cmd: the action we're trying to register
-    // - oldCmd: the action that already exists in our internal state. May be null.
-    // - maskingCmd: the masking action that already exists in our internal state. May be null.
-    void ActionMap::_TryUpdateName(const Model::Command& cmd, const Model::Command& oldCmd, const Model::Command& maskingCmd)
-    {
-        // Example:
-        //   { "name": "foo", "command": "copy" } --> we are setting a name, update oldCmd and maskingCmd
-        //   {                "command": "copy" } --> no change to name, exit early
-        const auto cmdImpl{ get_self<Command>(cmd) };
-        if (!cmdImpl->HasName())
-        {
-            // the user is not trying to update the name.
-            return;
-        }
-
-        // Update oldCmd:
-        //   If we have a Command in our _ActionMap that we're trying to update,
-        //   update it.
-        const auto newName{ cmd.Name() };
-        if (oldCmd)
-        {
-            // This command has a name, check if it's new.
-            if (newName != oldCmd.Name())
+            const auto cmdImpl{ get_self<implementation::Command>(cmd) };
+            if (cmd.Origin() == OriginTag::User && cmd.ID().empty())
             {
-                // The new name differs from the old name,
-                // update our name.
-                auto oldCmdImpl{ get_self<Command>(oldCmd) };
-                oldCmdImpl->Name(newName);
+                // the user did not define an ID for their non-nested, non-iterable, valid command - generate one for them
+                cmdImpl->GenerateID();
             }
-        }
 
-        // Update maskingCmd:
-        //   We have a Command that is masking one from a parent layer.
-        //   We need to ensure that this has the correct name. That way,
-        //    we can return an accumulated view of a Command at this layer.
-        //   This differs from oldCmd which is mainly used for serialization
-        //    by recording the delta of the Command in this layer.
-        if (maskingCmd)
-        {
-            // This command has a name, check if it's new.
-            if (newName != maskingCmd.Name())
+            // only add to the _ActionMap if there is an ID
+            if (auto cmdID = cmd.ID(); !cmdID.empty() && cmd.ActionAndArgs().Action() != ShortcutAction::Invalid)

```

Comment:
nit: may be cleaner to read if we do a early return for `cmdID.empty() || cmd.ActionAndArgs().Action() == ShortcutAction::Invalid` instead of another level of indenting

Reply:
agree. look at all that nesting

Reply:
This was actually just a mistake - was checking for `Invalid` twice! Removed the excess check and this should be much more readable now. Also we cannot early return on `cmdID.empty()` because if it's a user command we should generate an ID for it.